jbake-core/src/main/java/org/jbake/util/HtmlUtil.java

```diff
@@ -21,61 +21,79 @@ private HtmlUtil() {
+    {
         String siteHost = configuration.getSiteHost();
-        String uri = getDocumentUri(fileContents);
+        siteHost = StringUtils.appendIfMissing(siteHost, "/");
+
+        boolean prependSiteHost = configuration.getImgPathPrependHost();
+        boolean relativePointsToAssets = configuration.getRelativeImagePathsPointToAssets();
 
+        String dirUri = getDocumentUri(fileContents);
+
+        String htmlContent = fileContents.get(Attributes.BODY).toString();
         Document document = Jsoup.parseBodyFragment(htmlContent);
-        Elements allImgs = document.getElementsByTag("img");
 
+        Elements allImgs = document.getElementsByTag("img");
         for (Element img : allImgs) {
-            transformImageSource(img, uri, siteHost, prependSiteHost);
+            transformImageSource(img, dirUri, siteHost, prependSiteHost, relativePointsToAssets);
         }
 
-        //Use body().html() to prevent adding <body></body> from parsed fragment.
+        // Use body().html() to prevent adding <body></body> from parsed fragment.
         fileContents.put(Attributes.BODY, document.body().html());
     }
 
     private static String getDocumentUri(Map<String, Object> fileContents) {
-        String uri = fileContents.get(Attributes.URI).toString();
+        String dirUri = fileContents.get(Attributes.URI).toString();
 
         if (fileContents.get(Attributes.NO_EXTENSION_URI) != null) {
-            uri = fileContents.get(Attributes.NO_EXTENSION_URI).toString();
-            uri = removeTrailingSlash(uri);
+            dirUri = fileContents.get(Attributes.NO_EXTENSION_URI).toString();
+            dirUri = removeTrailingSlash(dirUri);
         }
 
-        if (uri.contains("/")) {
-            uri = removeFilename(uri);
+        if (dirUri.contains("/")) {
+            dirUri = removeFilename(dirUri);
         }
-        return uri;
+        return dirUri;
     }
 
-    private static void transformImageSource(Element img, String uri, String siteHost, boolean prependSiteHost) {
-        String source = img.attr("src");
+    private static void transformImageSource(Element img, String dirUri, String siteHost,
+                                             boolean prependSiteHost,
+                                             boolean relativePointsToAssets) {
+        String srcUrl = img.attr("src");
+        if (srcUrl.startsWith("http://") || srcUrl.startsWith("https://"))
+            return;
+
+        if (isRelativeToSourceMarkup(srcUrl, relativePointsToAssets))
+        {
+            // Image relative to current content is explicitly specified, lets add dir URI to it.
+            srcUrl = dirUri + srcUrl;
+        }
 
-        // Now add the root path
-        if (!source.startsWith("http://") && !source.startsWith("https://")) {
+        //source = StringUtils.removeStart(source, "/");
 
-            if (isRelative(source)) {
-                source = uri + source.replaceFirst("\\./", "");
```

Comment:
The removal of this line causes a few tests to fail.
If it really is unnecessary to replace paths starting with "./" you need to change the tests accordingly.
https://github.com/jbake-org/jbake/blob/master/jbake-core/src/test/java/org/jbake/util/HtmlUtilTest.java#L53 for example.

Reply:
Hi @ancho , that's something for discussion. I just rebased and din't have time to test. But IIRC, I needed some per-case way to turn on or off the site URL placing. The most natural way is to use such "neutral" prefix in `src` and give it a meaning. 
I will finish this and comment here.